typescript/src/deposit-sweep.ts

```diff
@@ -130,238 +150,297 @@ export async function assembleDepositSweepTransaction(
-    throw new Error("Deposit sweep transaction must have only one output")
+  // Sign the main UTXO input if there is main UTXO.
+  if (mainUtxo) {
+    const inputIndex = 0 // Main UTXO is the first input.
+    const previousOutput = Transaction.fromHex(mainUtxo.transactionHex).outs[
+      mainUtxo.outputIndex
+    ]
+
+    await signMainUtxoInput(
+      transaction,
+      inputIndex,
+      previousOutput.script,
+      previousOutput.value,
+      keyPair,
+      network
+    )
   }
 
-  // UTXOs must be mapped to deposits, as `fund` may arrange inputs in any
-  // order
-  const utxosWithDeposits: (UnspentTransactionOutput &
-    RawTransaction &
-    Deposit)[] = utxos.map((utxo, index) => ({
-    ...utxo,
-    ...deposits[index],
-  }))
-
-  for (let i = 0; i < transaction.inputs.length; i++) {
-    const previousOutpoint = transaction.inputs[i].prevout
-    const previousOutput = transaction.view.getOutput(previousOutpoint)
-    const previousScript = previousOutput.script
-
-    // P2(W)PKH (main UTXO)
-    if (previousScript.isPubkeyhash() || previousScript.isWitnessPubkeyhash()) {
-      await signMainUtxoInput(transaction, i, walletKeyRing)
-      continue
-    }
+  // Sign the deposit inputs.
+  for (let depositIndex = 0; depositIndex < deposits.length; depositIndex++) {
+    // If there is a main UTXO index, we must adjust input index as the first
+    // input is the main UTXO input.
+    const inputIndex = mainUtxo ? depositIndex + 1 : depositIndex
 
-    const utxoWithDeposit = utxosWithDeposits.find(
-      (u) =>
-        u.transactionHash.toString() === previousOutpoint.txid() &&
-        u.outputIndex == previousOutpoint.index
-    )
-    if (!utxoWithDeposit) {
-      throw new Error("Unknown input")
-    }
+    const utxo = utxos[depositIndex]
+    const previousOutput = Transaction.fromHex(utxo.transactionHex).outs[
+      utxo.outputIndex
+    ]
+    const previousOutputValue = previousOutput.value
+    const previousOutputScript = previousOutput.script
+
+    const deposit = deposits[depositIndex]
 
-    if (previousScript.isScripthash()) {
+    if (isP2SH(previousOutputScript)) {
       // P2SH (deposit UTXO)
-      await signP2SHDepositInput(transaction, i, utxoWithDeposit, walletKeyRing)
-    } else if (previousScript.isWitnessScripthash()) {
+      await signP2SHDepositInput(
+        transaction,
+        inputIndex,
+        deposit,
+        previousOutputValue,
+        keyPair
+      )
+    } else if (isP2WSH(previousOutputScript)) {
       // P2WSH (deposit UTXO)
       await signP2WSHDepositInput(
         transaction,
-        i,
-        utxoWithDeposit,
-        walletKeyRing
+        inputIndex,
+        deposit,
+        previousOutputValue,
+        keyPair
       )
     } else {
       throw new Error("Unsupported UTXO script type")
     }
   }
 
-  const transactionHash = TransactionHash.from(transaction.txid())
+  const transactionHash = TransactionHash.from(transaction.getId())
```

Comment:
Just making sure: `getId` returns the hash computed using the standard serialization format `[version][inputs][outputs][locktime]` and in the RPC (block explorer) byte order right?

Reply:
Yes, we check the transaction hash (which built from `getId`) in our unit tests.